azure-pipelines.yml

```diff
@@ -0,0 +1,343 @@
+jobs:
+- job: 'Ubuntu16All'
+  condition: or(eq(variables['Build.SourceBranchName'], 'master'), eq(variables['Build.Reason'], 'PullRequest'))
```

Comment:
Why is this needed? can't we use the following syntax:
```
# Only build PRs that target master
pr:
  # Additional pushes to a PR cancel the current build
  autoCancel: true 
  branches:
    include: 
      - master
```

Reply:
Unfortunately, we cannot. The `pr` and `trigger` keywords operate at the build granularity, here we need to enable or disable specific jobs. The relevant examples are here: 
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml